flask_resources/parsers/url_args.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Flask-Resources is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Library for easily implementing REST APIs."""
+
+from flask import request
+from functools import wraps
+
+from .base import RequestParser, select_request_parser
+from ..context import resource_requestctx
+
+
+def url_args_parser(f):
+    """Decorator that parses the URL query string for the view.
+
+    NOTE: Flask captures the query string on ``request.args`` ergo the compact name.
+    """
+
+    @wraps(f)
+    def wrapper(self, *args, **kwargs):
+        """Wrapping method.
+
+        NOTE: To extract request args from any endpoint and not repeat ourselves,
+              extraction is done here and inspects self + request.method to get
+              endpoint.
+
+        :params self: Item/List/SingletonView instance
+        """
+        # TODO: Create interfaces to not have to break Demeter's law here
+        parser = select_request_parser(
+            self.resource_method,
+            self.resource.config.request_url_args_parser,
+            URLArgsParser
+        )
+        resource_requestctx.url_args = parser.parse()
+        return f(self, *args, **kwargs)
+
+    return wrapper
+
+
+class URLArgsParser(RequestParser):
+    """URL query string parser."""
+
+    def __init__(self, argmap=None, allow_unknown=True):
+        """Constructor.
+
+        :params argmap: dict or Schema of URL querystring arguments
+        :params allow_unknown: bool allow unknown URL querystring arguments or not
+        """
+        super(URLArgsParser, self).__init__(argmap, allow_unknown,
+                                            locations=("querystring",))
+
+    @property
+    def raw_args(self):
```

Comment:
What is this method used for?

Reply:
See [comment](https://github.com/inveniosoftware/flask-resources/pull/83/files/c6446f24ef4b45a87e98658c90efd7e05b624ada#diff-4a46d6efa6fb0d76ee23076ce4929d42R59)